### ISSUE TITLE
Add an optimzer based on genetic algorithm

### ISSUE THREAD

**Issue by jaykay9999** [keras-team/keras#16549]
If you open a GitHub issue, here is our policy:

It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
The form below must be filled out.

**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**System information**.

TensorFlow version (you are using):

- Latest Version

Are you willing to contribute it (Yes/No) :

- Yes
- 
**Describe the feature and the current behavior/state**.

Describe the feature clearly here. Be sure to convey here why the requested feature is needed. Any brief description about the use-case would help.

A genetic algorithm based optimizer that tries to find the best set of weights for large training data sets. I have been working with the genetic algorithm and it is simply an algorithm to find patterns for example the shortest path problem, but in this case it will be the pattern of the best vector of weights.

**Will this change the current api? How?**

No

**Who will benefit from this feature?**

Whenever the input layer is huge (more than a million input) and it takes alot of time to find a good set of weights.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no):

Yes

- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

-There are 3 different things in this optimizer:

Phenotype : which is a weight
Dna : which is a sum of phenotypes which translates to a vector of weights.
Population : A group of Dna's. 

population 1: At first, the network will randomly generate a population of let's say 100 random DNA's, (the user inputs the number), then an accuracy will be calculated for each vector of weights, The 10 fittest (top 10 accuracies) Dna's will crossover (crossover means two dna's will result in a child Dna that has let's say 50% of the weights of each parent). 

Population 2: Will have 100 dna that are all the childs of the 10 fittest dna from population 1, the same thing will occur again to result in population 3 then 4 and it goes on (the user inputs when it should stop)

There is also a mutation function in which there is a small chance that whenever two DNA's crossover, a weight in their child dna will randomly change to keep the possiblity that it might find better patterns. 

I'm open to further elaborate my solution and answer questions regarding this feature :)

**Comment by sushreebarsa** [comment]
@jaykay9999 Could you please elaborate about your feature and please specify the use cases for this feature. Thank you!
